src/main/java/Parser.java

```diff
@@ -0,0 +1,136 @@
+            }
+        } else if (command.startsWith("find")) {
+            String keyword = command.substring(5).trim();
+            taskList.findTasks(keyword);
+        } else if (command.equals("bye")) {
+            Ui.showGoodbyeMessage();
+            System.exit(0);
+        } else {
+            System.out.println("____________________________________________________________");
+            System.out.println("     OOPS!!! Unknown request :(");
+            System.out.println("____________________________________________________________");
+        }
+    }
+
+    /**
+     * Handles the creation of a ToDo task.
+     *
+     * @param command The user input command.
+     * @param taskList The TaskList to store the task.
+     */
+    private static void handleTodo(String command, TaskList taskList) {
+        String description = command.substring(4).trim();
+
+        if (description.isEmpty()) {
+            ErrorHandler.handleEmptyDescription();
+            Nanat.startCommandLoop(taskList);
+        }
+
+        while(description.charAt(0) == ' ') {
+            description.substring(1).trim();
+        }
+
+        int num = taskList.numOfTasks();
+        Ui.showAddedMessage(1, description, num, taskList);
+    }
+
+    /**
+     * Handles the creation of a Deadline task.
+     *
+     * @param command The user input command.
+     * @param taskList The TaskList to store the task.
+     */
+    private static void handleDeadline(String command, TaskList taskList) {
+        String description = command.substring(8).trim();
+        String[] parts = description.split(" /by ", 2);
+
+        if (description.isEmpty() || !description.contains("by")) {
+            ErrorHandler.handleEmptyDeadline();
+            Nanat.startCommandLoop(taskList);
+        }
+
+        while(description.charAt(0) == ' ') {
+            description.substring(1).trim();
+        }
+
+        if (parts[0].trim().isEmpty() || parts[1].trim().isEmpty()) {
+            ErrorHandler.handleEmptyDeadline();
+            Nanat.startCommandLoop(taskList);
+        }
+
+        int num = taskList.numOfTasks();
+        Ui.showAddedMessage(2, description, num, taskList);
+    }
+    /**
+     * Handles the creation of an Event task.
+     *
+     * @param command The user input command.
+     * @param taskList The TaskList to store the task.
+     */
+    private static void handleEvent(String command, TaskList taskList) {
+        String description = command.substring(5).trim();
+
+        if (description.isEmpty() || !description.contains("from") || !description.contains("to")) {
+            ErrorHandler.handleEmptyEvent();
+            Nanat.startCommandLoop(taskList);
+        }
+
+        while(description.charAt(0) == ' ') {
+            description.substring(1).trim();
+        }
+
+        String[] parts = description.split(" /from ", 2);
+        String[] timeParts = parts[1].split(" /to ", 2);
```

Comment:
consider extracting low-level string operations to separate methods for better SLAP